lib/scripts/repo_export_in_batches.rb

```diff
@@ -0,0 +1,44 @@
+# Before you run this script empty the directory tmp/export/
+#
+# Run this script with the following:
+#   ./bin/rails r lib/scripts/repo_export_in_batches > "$(date +'%Y-%m-%d-%H%M%S')_repo_export_custom.log"
+#
+puts "---------\nBeginning repo export at #{Time.now} #{Time.zone}\n---------"
+
+# set classes to use for export
+record_converter = {model_class: GenericFile, converter_class: InvenioRdmRecordConverter}
+collection_converter = {model_class: "", converter_class: ""}
+
+# validate classes
+puts "---------\nValidating converter classes\n---------"
+raise(RegistryError, "Model (#{model_class.name}) for conversion must be an ActiveFedora::Base") unless record_converter[:model_class].ancestors.include?(ActiveFedora::Base)
+# TODO: Implement collection conversion
+# raise(RegistryError, "Converter (#{converter_class.name}) for conversion must be an Sufia::Export::Converter") unless converter[:converter_class].ancestors.include?(Sufia::Export::Converter)
+
+puts "---------\nBeginning file export at #{Time.now} #{Time.zone}\n---------"
+generic_file_conversion_count = 0
+GenericFile.find_each do |gf|
+  converted_record_object = record_converter[:converter_class].new(gf)
+
+  puts "---------\nFile has id: #{gf.id}\n---------"
+  file_path = "tmp/export/#{record_converter[:model_class].name.underscore}_#{gf.id}.json"
+  File.write(file_path, converted_record_object.to_json(pretty: true))
+
+  # increment and put object out of scope
+  generic_file_conversion_count += 1
+  converted_record_object = nil
```

Comment:
Just curious about this: why is that necessary? Isn't the line https://github.com/galterlibrary/digital-repository/pull/961/files#diff-ec7071bf61726d0fda71d83e812c135e96ef7c7fc6ce7ad9bf5460a5e43417bbR21 above getting rid of the only reference to the "converted record object"?

Reply:
Yeah, this was done in error. I read some post about triggering the garbage collector by removing all references to objects, but then I applied it here where it doesn't really fit.